app/shared/user-agent-client.js

```diff
@@ -37,98 +35,81 @@ function uaRequest(service, options) {
   });
 }
 
-class UserAgent extends EventEmitter {
-  constructor(ws, clientCount) {
+class UserAgentClient extends EventEmitter {
+  constructor(url = endpoints.UA_SERVICE_WS) {
     super();
-
-    this.ws = ws;
-    this.clientCount = clientCount;
-
-    this.ws.on('message', (data) => {
-      data = JSON.parse(data);
-      const message = data.message;
-      delete data.message;
-      this.emit(message, data);
-    });
+    this.url = url;
   }
 
-  startSession() {
-    return uaRequest('/session/start', {
+  startSession(url = endpoints.UA_SERVICE_HTTP) {
+    return uaRequest(url, '/session/start', {
```

Comment:
Random thought, rather than needing all callers to await connect() first we could just do it in here.